### PR TITLE
OSD-19807 : onboard hypershift logging operator codecov job

### PR DESCRIPTION
**Pull request by samanthajayasinghe**
# Why 
This PR onboard codecov job to hypershift-logging-operator repo

### CODE REVIEW

**Comment by xiaoyu74** [comment]
Need to fix the merge conflicts.

**Comment by samanthajayasinghe** [comment]
@xiaoyu74 Rebased the PR .. pls have a look again

File: ci-operator/config/openshift/hypershift-logging-operator/openshift-hypershift-logging-operator-main.yaml [hunk 1]
```diff
@@ -11,6 +11,25 @@ resources:
...
+- as: publish-coverage
+  commands: |
+    export CODECOV_TOKEN=$(cat /tmp/secret/CODECOV_TOKEN)
+    make coverage
+  container:
+    from: src
+  postsubmit: true
+  secret:
+    mount_path: /tmp/secret
+    name: hypershift-logging-operator-codecov-token
```
Comment (xiaoyu74) [inline ci-operator/config/openshift/hypershift-logging-operator/openshift-hypershift-logging-operator-main.yaml:32]: Have you already update the `hypershift-logging-operator-codecov-token` token in a specific path in Vault?

Comment (samanthajayasinghe) [reply ci-operator/config/openshift/hypershift-logging-operator/openshift-hypershift-logging-operator-main.yaml:32]: Thanks for the heads up @xiaoyu74 , I'll grab someone who has permission to copy the token

**Pull request merged**